### ISSUE TITLE
[pthreads:x86-windows] build failure

### ISSUE THREAD

**Issue by gvanem** [microsoft/vcpkg#17075]
I needed to rebuild my `cairo` package since a failure during `vcpkg upgrade` deleted that and several others!
But ran into problems with the obviously dependant package `pthreads`.

**Host Environment**
- OS: Windows-10
- Compiler: MSVC-2019, 19.28.29912 for x86

**To Reproduce**
`vcpkg install cairo:x86-windows --debug`

**Failure logs** 
All went well until building `x86-windows-dbg`. Here: [nmake-build-x86-windows-debug-err.log](https://github.com/microsoft/vcpkg/files/6254050/nmake-build-x86-windows-debug-err.log)

```
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: F:/gv/vc_2019/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x86/nmake.exe /G -f Makefile all install 
DESTROOT="F:\\gv\\VC_project\\VCPKG\\packages\\pthreads_x86-windows\\debug"
    Working Directory: F:/gv/VC_project/VCPKG/buildtrees/pthreads/src/49e541b66c-757bea5a70.clean
    Error code: 2
    See logs for more information:
      F:\gv\VC_project\VCPKG\buildtrees\pthreads\nmake-build-x86-windows-debug-out.log
      F:\gv\VC_project\VCPKG\buildtrees\pthreads\nmake-build-x86-windows-debug-err.log
```

Seems my shell 4NT (and `cmd` too) have problem at the `mkdir "foo"\bin`. Maybe `"foo"` does not exist?

(the given variable `DESTROOT="..."` pasted with `\bin` creates this path).

How/who creates that?

**Comment by Cheney-W** [comment]
Thanks for reporting this issue. 
I tried to reproduce it with the latest vcpkg: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e and VS2019 16.9.3, and I could install cairo:x86-windows successfully.

**Comment by Cheney-W** [comment]
@gvanem Does this issue still occur when you install this port?

**Comment by gvanem** [comment]
> Does this issue still occur when you install this port?

Yes, the generated/patched Makefile for debug-mode is plain wrong. Hence my shell (4NT) complains; see below.
How do I avoid generating a debug-version of pthreads?

Full output of `vcpkg install pthreads:x86-windows --debug`, [here](https://github.com/microsoft/vcpkg/files/6513462/build-log.txt) (slightly edited)
And last line of `nmake-build-x86-windows-debug-out.log` :
```
if not exist "F:\gv\VC_project\VCPKG\packages\pthreads_x86-windows\debug"\bin mkdir "F:\gv\VC_project\VCPKG\packages\pthreads_x86-windows\debug"\bin
```
Illegal shell-syntax!

And last line of `nmake-build-x86-windows-debug-err.log` :
```
4NT: (Sys) File not found.
 "F:\gv\VC_project\VCPKG\packages\pthreads_x86-windows\debug\bin"
NMAKE : fatal error U1077: 'if' : return code '0x2'
Stop.
```
